src/SqlBindingUtilities.cs

```diff
@@ -290,5 +290,51 @@ internal static bool IsFatalSqlException(this Exception e)
             }
             return true;
         }
+
+        /// <summary>
+        /// Query the ServerProperty('EngineEdition') for logging the Server Type for Telemetry
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Throw if an error occurs while querying the engine edition</exception>
+        public static async Task<string> GetSqlServerEdition(SqlConnection connection, ILogger logger, CancellationToken cancellationToken)
+        {
+            string selectServerEdition = $"SELECT ServerProperty('EngineEdition')";
```

Comment:
nit: Built in functions are all upper cased by convention
```suggestion
            string selectServerEdition = $"SELECT SERVERPROPERTY('EngineEdition')";
```